sys/include/suit/coap.h

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2019 Kaspar Schleiser <kaspar@schleiser.de>
+ *               2019 Inria
+ *               2019 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+/**
+ * @defgroup    sys_suit SUIT secure firmware updates
+ * @ingroup     sys
+ * @brief       SUIT secure firmware updates
+ *
+ * @{
+ *
+ * @brief       SUIT CoAP helper API
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ */
+
+#ifndef SUIT_COAP_H
+#define SUIT_COAP_H
+
+#include "net/nanocoap.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief    Start SUIT CoAP thread
+ */
+void suit_coap_run(void);
+
+/**
+ * @brief SUIT CoAP endpoint entry.
+ *
+ * In order to use, include this header, then add SUIT_COAP_SUBTREE to the nanocoap endpoint array.
+ * Mind the alphanumerical sorting!
+ *
+ * See examples/suit_update for an example.
+ */
+#define SUIT_COAP_SUBTREE \
+    { \
+        .path="/suit/", \
+        .methods=COAP_MATCH_SUBTREE | COAP_METHOD_GET | COAP_METHOD_POST | COAP_METHOD_PUT, \
+        .handler=coap_subtree_handler, \
+        .context=(void*)&coap_resource_subtree_suit \
+    }
+
+
+/* this is internal code that will go soon, thus skip Doxygen checks */
```

Comment:
I think this needs a better justification.

Reply:
added some prose, please check

Reply:
Perfect